### PR TITLE
Quickstart wordpress

### PR DESCRIPTION
**Pull request by sixhobbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205273462533574

### CODE REVIEW

**Comment by alex-fusionauth** [comment]
Moved to new repo and correct perms
https://github.com/FusionAuth/fusionauth-quickstart-wordpress-web

**Comment by alex-fusionauth** [comment]
@sixhobbits when trying to run this on Mac with M2
```
╰─ docker compose up -d                     
[+] Running 0/1
 ⠧ wpdb Pulling                                                                                                      0.7s 
no matching manifest for linux/arm64/v8 in the manifest list entries
```
I am not great at docker but I followed this
https://stackoverflow.com/questions/65456814/docker-apple-silicon-m1-preview-mysql-no-matching-manifest-for-linux-arm64-v8

When I update to 
```
  wpdb:
    image: mysql:5.7
    platform: linux/x86_64
```
everything works, but I am wondering if this will have an impact on windows/other systems for users. Is there a way to make this more generic?

**Review by alex-fusionauth — CHANGES_REQUESTED** [review]
Just a couple of small grammatical things to look at.

One issue with the Docker compose on Mac M2.

File: astro/src/content/quickstarts/quickstart-wordpress-web.mdx [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+- <Field label="End Session Endpoint URL"/> - `http://localhost:9011/oauth2/logout`
+- <Field label="Nickname Key"/> - `sub`
+- <Field label="Display Name Formatting"/> - `{email}` (this is what will be displayed to the user in the {frontmatter.technology} admin screen)
+- Enable <Uielement label="Link Existing Users"/> - this marks a user in your local {frontmatter.technology} database who has the same email in FusionAuth as the same user, instead of duplicating them.
+
+Click <Uielement label="Save Changes"/>.
+
+<img src="/img/quickstarts/quickstart-wordpress-web/oidc-config.png" alt="Configure FusionAuth"/>
+
+Now if you browse to http://localhost:3000/wp-login.php, you can click <Uielement label="Login with OpenID Connect"/> and log in on FusionAuth with email `admin@example.com` and password `password`.
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-wordpress-web.mdx:193]: @alex-fusionauth this I think is as result of the astro preview also using port 3000. Can you confirm if you were running the astro preview at the same time.

[the remark this answers is not in the document]
Comment (alex-fusionauth) [reply astro/src/content/quickstarts/quickstart-wordpress-web.mdx:193]: I think it was the amount of cookies running against it. As I had moved to using Astro on 3001.

File: astro/src/content/quickstarts/quickstart-wordpress-web.mdx [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+You now have three blank pages.
+
+<img src="/img/quickstarts/quickstart-wordpress-web/home.png" alt="Add page"/>
+
+Browse to <Breadcrumb label="Settings -> Reading"/>. Change <Field label="Your homepage displays"/> to `A static page` and under <Field label="Homepage"/> select `Home`. Save your changes. Now http://localhost:3000 points to your homepage.
+
+<img src="/img/quickstarts/quickstart-wordpress-web/reading.png" alt="Homepage"/>
+
+Finally, browse to <Breadcrumb label="Appearance -> Editor -> Templates -> Blank"/>. Here you are editing the page style template for all three of your new pages. Click on the paragraph (`This is the Post Content block`) twice to select it. Under <Breadcrumb label="Layout"/> on the right, disable <Uielement label="Inner blocks use content width"/>. Click <Uielement label="Save"/> and confirm. This changes each page to use the full browser width.
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-wordpress-web.mdx]: added an instruction which mentions opening the settings menu

**Comment by rideam** [comment]
@alex-fusionauth  I have opened a PR https://github.com/FusionAuth/fusionauth-quickstart-wordpress-web/pull/1 which specifies the platform for the docker images.  Specifying the platform just forces docker to use emulation instead of the architecture of the host so the solution runs fine on other platforms.

File: astro/src/content/quickstarts/quickstart-wordpress-web.mdx [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+<img src="/img/quickstarts/quickstart-wordpress-web/media.png" alt="Add image"/>
+
+Click on the image once it has been uploaded. Copy the "File URL" to use later. It should look like `http://localhost:3000/wp-content/uploads/2023/09/money.jpg`.
+
+### Add Custom Page Shortcodes
+
+You'll now create three shortcodes with the HTML, CSS, and JavaScript for each page.
+
+Browse to <Breadcrumb label="Wow Plugins -> WP Coder"/>. Click <Uielement label="Add New"/>. In the title box at the top, enter `home`. In the <Field label="HTML code"/> box, copy the file contents from `complete-application/home.html`. In the `<img src` at the bottom of this code, enter the file URL for the money image you uploaded. Click the <Breadcrumb label="CSS Code"/> tab, and copy the file contents from `changebank.css`. Click <Uielement label="Save Changes"/>.
+
```
Comment (mooreds) [inline astro/src/content/quickstarts/quickstart-wordpress-web.mdx:265]: Please import the files so it is easy to copy paste.

**Review by mooreds — CHANGES_REQUESTED** [review]
A couple of issues:

* The readme in github repo is not correct, doesn't follow the standards for all the other quickstart repos. See https://github.com/fusionauth/fusionauth-quickstart-python-flask-web

In both the complete application and working through the quickstart, I sometimes get an error: `Your browser sent a request that this server could not understand. Size of a request header field exceeds server limit.`Opening it up in an incognito window solved it, so can you add that to the troubleshooting section?

File: astro/src/content/quickstarts/quickstart-wordpress-web.mdx [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+- <Field label="Client Secret Key"/> - `super-secret-secret-that-should-be-regenerated-for-production`
+- <Field label="OpenID Scope"/> - `openid`
+- <Field label="Login Endpoint URL"/> - `http://localhost:9011/oauth2/authorize`
+- <Field label="Userinfo Endpoint URL"/> - `http://fusionauth:9011/oauth2/userinfo`
+- <Field label="Token Validation Endpoint URL"/> - `http://fusionauth:9011/oauth2/token`
+- <Field label="End Session Endpoint URL"/> - `http://localhost:9011/oauth2/logout`
+- <Field label="Nickname Key"/> - `sub`
+- <Field label="Display Name Formatting"/> - `{email}` (this is what will be displayed to the user in the {frontmatter.technology} admin screen)
+- Enable <Uielement label="Link Existing Users"/> - this marks a user in your local {frontmatter.technology} database who has the same email in FusionAuth as the same user, instead of duplicating them.
+
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-wordpress-web.mdx:191]: added the note

File: astro/src/content/quickstarts/quickstart-wordpress-web.mdx [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+Click <Uielement label="Save Changes"/>.
+
+<img src="/img/quickstarts/quickstart-wordpress-web/oidc-config.png" alt="Configure FusionAuth"/>
+
+Now if you browse to http://localhost:3000/wp-login.php, you can click <Uielement label="Login with OpenID Connect"/> and log in on FusionAuth with email `admin@example.com` and password `password`.
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-wordpress-web.mdx]: changed to use `richard@example.com`

File: astro/src/content/quickstarts/quickstart-wordpress-web.mdx [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+On the success page, click <Uielement label="Log In"/>. On the login page, enter the username `admin` and password `password` and click <Uielement label="Log In"/>. (You can return to this page later by browsing to http://localhost:3000/wp-login.php.)
+
+### Create Placeholder Pages
+
+Before you do any FusionAuth work, you'll create three placeholder pages:
+- A home page that anonymous users can see.
+- A make change page that only logged in users can see.
+- An account page that only logged in users can see.
+
+Under <Breadcrumb label="Settings -> Permalinks"/>, change <Field label="Permalink structure"/> to `Post name` and click <Uielement label="Save Changes"/>. This sets page URLs to be more human-readable than the default numbered URLs.
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-wordpress-web.mdx]: This is fixed now. The implementation of the Breadcrumb was changed to not use the `label` attribute and hence it was no longer rendering properly.

File: astro/src/content/quickstarts/quickstart-wordpress-web.mdx [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+
+<img src="/img/quickstarts/quickstart-wordpress-web/for-logged-in.png" alt="Page settings"/>
+
+### Send User To Account After Log In
+
+The final improvement you can make to the login process is to send the user directly to their account page instead of the dashboard after login.
+
+Search for and install the plugin "Remove Dashboard Access" by TrustedLogin.
+
+Browse to <Breadcrumb label="Settings -> Dashboard Access"/>. Change the value of <Field label="Redirect URL"/> to `http://localhost:3000/account`. Click <Uielement label="Save Changes"/>.
```
[the remark this answers is not in the document]
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-wordpress-web.mdx]: with reference to the image you attached , logging in with richard gives you access to the account page (which would be blank at this point) but you will not be able to access the administrative menu

shots

File: astro/src/content/quickstarts/quickstart-wordpress-web.mdx (hunk 3 above)
Comment (rideam) [reply astro/src/content/quickstarts/quickstart-wordpress-web.mdx:275]: code imported

**Comment by rideam** [comment]
@mooreds I opened a PR here https://github.com/FusionAuth/fusionauth-quickstart-wordpress-web/pull/4 to update the link to the changebank logo

**Review by mooreds — APPROVED** [review]

**Pull request merged**